### PR TITLE
refactors behavior_options.py

### PR DESCRIPTION
**Pull request by NishanthJKumar**
* splits off functions that return policy function so integrating option model will be significantly easier
* includes some helpful status prints within option execution
* renames behavior_env attr of BehaviorEnv to igibson_behavior_env to be clearer

### CODE REVIEW

File: src/envs/behavior_options.py [hunk 1]
```diff
@@ -588,61 +662,12 @@ def grasp_obj_at_pos(
...
-
-        return ret_action, done_bit
-
     p.restoreState(state)
     p.removeState(state)
 
-    return graspObjectOption
+    print(f"PRIMITIVE: grasp {obj.name} success! Plan found with " +
+          f"continuous params {grasp_offset}. Executing now.")
+    return create_grasp_policy(plan)
```
Comment (ronuchit) [inline src/envs/behavior_options.py:670]: i see, so the idea will be if we're in model mode, we shouldn't use this policy, but rather just extract the final state from the plan? and teleport to it?

is that final state from the plan the same as this variable `state`? or is `state` the state before we screw things up with all this planning?

File: src/envs/behavior_options.py [hunk 2]
```diff
@@ -651,7 +676,7 @@ def place_obj_plan(
     original_state: int,
     place_rel_pos: Array,
     rng: Optional[Generator] = None,
-) -> List[List[float]]:
+) -> Optional[List[List[float]]]:
```
Comment (ronuchit) [inline src/envs/behavior_options.py:679]: why this change?

**Review by ronuchit — APPROVED** [review]

File: src/envs/behavior_options.py (hunk 2 above)
Comment (NishanthJKumar) [reply src/envs/behavior_options.py:679]: turns out it actually was always optional and this was just a bug (caught thanks to the refactor)!

Comment (NishanthJKumar) [reply src/envs/behavior_options.py:679]: (it's optional because the plan might be None)

File: src/envs/behavior_options.py (hunk 1 above)
Comment (NishanthJKumar) [reply src/envs/behavior_options.py:670]: so the idea in model mode will be actually that we'll call a function `create_grasp_model` similar to `create_grasp_policy` that will take in the `plan` and then teleport the world to the right end configuration (not exactly, since we'll be doing some extra legwork to make it more realistic so that we can be sure that if the option model works, the actual option policy execution will also work)

`state` is a saved state from before we screw things up with planning that we can use to reset the world before returning our function (that's why there's the `p.restoreState(state)` line before the `return`). The plan is actually a series of waypoints that might be for the hand, or for the body depending on what planning function is called, and so we can't simply return the last element, we have to do some work to teleport the robot (and objects it might be holding, etc.)
